haraka provision error

### Expected behavior
Should be provisioned.

### Observed behavior
------------------------------------------------------------------------
jexec stage npm install --production -g haraka/Haraka ws express
/usr/local/bin/spf -> /usr/local/lib/node_modules/Haraka/bin/spf
/usr/local/bin/dkimverify -> /usr/local/lib/node_modules/Haraka/bin/dkimverify
/usr/local/bin/haraka_grep -> /usr/local/lib/node_modules/Haraka/bin/haraka_grep
npm ERR! path /usr/local/lib/node_modules/Haraka/bin/haraka
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall chmod
npm ERR! enoent ENOENT: no such file or directory, chmod '/usr/local/lib/node_modules/Haraka/bin/haraka'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2017-12-16T09_52_25_079Z-debug.log

### Steps to reproduce
Fresh installation
run: provision haraka

###
FreeBSD 11.1-RELEASE-p4 FreeBSD 11.1-RELEASE-p4 #0: Tue Nov 14 06:12:40 UTC 2017     root@amd64-builder.daemonology.net:/usr/obj/usr/src/sys/GENERIC  amd64

--- a/provision-haraka.sh
+++ b/provision-haraka.sh
@@ -13,18 +13,20 @@ HARAKA_CONF="$ZFS_DATA_MNT/haraka/config"
 install_haraka()
 {
 	tell_status "installing node & npm"
-	stage_pkg_install npm-node8 gmake python || exit
+	stage_pkg_install npm-node8 gmake python git-lite || exit
+	stage_exec npm install -g --only=prod node-gyp || exit
 
 	tell_status "installing Haraka"
-	stage_exec pkg install -y git-lite
+	stage_exec git clone --depth=1 https://github.com/haraka/Haraka.git /tmp/Haraka
+	stage_exec npm set user 0
+	stage_exec npm set -g unsafe-perm true
+	stage_exec npm install -g --only=prod /tmp/Haraka || exit
 
-	stage_exec npm install --production -g haraka/Haraka ws express || exit
-
-	local _plugins="haraka-plugin-log-reader"
+	local _plugins="ws express haraka-plugin-log-reader"
 	for _p in known-senders aliases; do
 		_plugins="$_plugins haraka-plugin-$_p"
 	done
-	stage_exec bash -c "cd /data && npm install --production $_plugins"
+	stage_exec bash -c "cd /data && npm install --only=prod $_plugins"
 }
 
 install_geoip_dbs()

